### PR TITLE
Add test using verify checksum on non yum repo in hammer

### PR DESCRIPTION
**Pull request by sambible**
Simple modification of an existing test, to cover a BZ case seen recently 
https://bugzilla.redhat.com/show_bug.cgi?id=2161209

### CODE REVIEW

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py::test_verify_checksum_non_yum_repo

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k 'test_verify_checksum_non_yum_repo'

File: tests/foreman/cli/test_repository.py [hunk 1]
```diff
@@ -908,6 +908,40 @@ def test_positive_synchronize_docker_repo(self, repo, module_product, module_org
...
+                    'content-type': 'docker',
+                    'docker-upstream-name': CONTAINER_UPSTREAM_NAME,
+                    'name': valid_docker_repository_names()[0],
+                    'url': CONTAINER_REGISTRY_HUB,
+                }
+            ]
+        ),
+        indirect=True,
+    )
+    def test_verify_checksum_non_yum_repo(self, repo):
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_repository.py:926]: Mentioned BZ says `Verify Content Checksum, on container repositories`, and as test verifies only docker repository
```suggestion
    def test_verify_checksum_container_repo(self, repo):
```

File: tests/foreman/cli/test_repository.py [hunk 2]
```diff
@@ -908,6 +908,40 @@ def test_positive_synchronize_docker_repo(self, repo, module_product, module_org
...
+        """Check if Verify Content Checksum can be run on non yum/deb repos
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_repository.py:927]:
```suggestion
        """Check if Verify Content Checksum can be run on container repository
```

File: tests/foreman/cli/test_repository.py [hunk 3]
```diff
@@ -908,6 +908,40 @@ def test_positive_synchronize_docker_repo(self, repo, module_product, module_org
...
+        :id: c8f0eb45-3cb6-41b2-aad9-52ac847d7bf8
+
+        :parametrized: yes
+
+        :customerscenario: true
+
+        :expectedresults: Docker repository is created, and can be synced with
+            validate-contents set to True
+
+        :BZ: 2161209
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_repository.py:938]:
```suggestion
        :expectedresults: Docker repository is created, and can be synced with
            validate-contents set to True

        :BZ: 2161209
        
        :customerscenario: true
```

File: tests/foreman/cli/test_repository.py [hunk 4]
```diff
@@ -908,6 +908,40 @@ def test_positive_synchronize_docker_repo(self, repo, module_product, module_org
...
+        """
+        assert repo['sync']['status'] == 'Not Synced'
+        Repository.synchronize({'id': repo['id'], 'validate-contents': 'true'})
+        new_repo = Repository.info({'id': repo['id']})
+        assert new_repo['sync']['status'] == 'Success'
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_repository.py:943]: Do we've add any additional check that could be added to verify `'validate-contents': 'true'`, other than sync status?

File: tests/foreman/cli/test_repository.py [hunk 5]
```diff
@@ -908,6 +908,40 @@ def test_positive_synchronize_docker_repo(self, repo, module_product, module_org
             ]
         )
 
+    @pytest.mark.tier2
+    @pytest.mark.parametrize(
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_repository.py:912]: Mentioned BZ also has one upgrade related case attached, so it'd be better to cover this for upgrades
```suggestion
    @pytest.mark.tier2
    @pytest.mark.upgrade
    @pytest.mark.parametrize(
```

**Review by Gauravtalreja1 — APPROVED** [review]
Ack, with few suggestions

**Review by Griffin-Sullivan — APPROVED** [review]
Ack pending suggestions

**Review by ColeHiggins2 — APPROVED** [review]
Ack

**Review by synkd — APPROVED** [review]

File: tests/foreman/cli/test_repository.py (hunk 4 above)
Comment (sambible) [reply tests/foreman/cli/test_repository.py:943]: I can't think of a way. I think this is fine, as if it wasn't possible, you would get an error code returned from the command.

**Comment by sambible** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_repository.py -k 'test_verify_checksum_non_yum_repo'

**Pull request merged**